packages/headless/src/api/analytics/shared-analytics.ts

```diff
@@ -0,0 +1,9 @@
+import {ConfigurationSection} from '../../state/state-sections';
+
+export const getLanguage = (state: ConfigurationSection) => {
+  const langKey = state.configuration.search.locale.split('-')[0];
+  if (!langKey || langKey.length !== 2) {
```

Comment:
Does the check prevent setting languages without the second part, such as `fr` ? `langKey` is 1 in that case

Reply:
Yeah, normally for `locale` (so, the state.configuration.search.locale parameter where it comes from) it needs to follow a specific format.

I added a documentation comment about this, but reposting here: 


```
The locale of the current user. Must comply with IETF’s BCP 47 definition: https://www.rfc-editor.org/rfc/bcp/bcp47.txt.
```

It's validated by a regex that's a bit too simplistic, but that should be "good enough".

```
export const localeValidation = new StringValue({
  emptyAllowed: false,
  required: false,
  regex: /[a-z]{2}-[A-Z]{2}/,
});
```

I'm saying too simplistic, because the actual regex that would strictly validate the RFC is something like 20 lines long...

Anyway 🤷 ... the regex should ensure that no one can just set `fr` for their locale.

Reply:
Should the conversion from BCP47 to the analytics format be done inside `headless`?

I think it could be better to have the transformation occur inside `coveo.analytics` so that other projects can benefit from the code. Also, if one day the analytics team want to change to BCP format, or have both, only `coveo.analytics` would need to change.

Reply:
So the BCP47 stuff is strictly a search API concern, and the language key for analytics is strictly an analytics concern.

So it would not 100% make sense to do that inside coveo.analytics, I think.

Reply:
Ok, I can see this point. My suggestion assumes coveo.analytics is the analytics lib for the searchapi, but since it's intended for any coveo api, maybe the way things are now is best.